internal/devices/reinstall.go

```diff
@@ -0,0 +1,65 @@
+// Copyright © 2018 Jasmin Gacic <jasmin@stackpointcloud.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package devices
+
+import (
+	"fmt"
+
+	"github.com/packethost/packngo"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Reinstall() *cobra.Command {
+	var (
+		id string
+
+		operatingSystem string
+		preserveData    bool
+	)
+
+	reinstallDeviceCmd := &cobra.Command{
+		Use:   `reinstall -d <device-id>`,
+		Short: "Reinstalls a device.",
+		Long:  "Reinstalls the provided device. The ID of the device to reinstall is required.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			request := packngo.DeviceReinstallFields{
+				OperatingSystem: "",
```

Comment:
It looks like we missed the `operatingSystem` here and reused `preserveData` for `DeprovisionFast` (🤔 maybe that combination makes sense?).  There are a few other flags we could have exposed.

Reply:
Should be fixed by https://github.com/equinix/metal-cli/pull/238. PreserveData and DeprovisionFast are tied to the same option because it doesn't make sense to have one without the other. osie will only skip wiping disks if both are set.